Eastern-Research-Group/mywaterway
# Feature/390 more surrounding layers

## Related Issues:
* [HMW-390](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-390)

## Main Changes:
* Added a "Surrounding Features" widget that controls visibility of surrounding features for multiple layers.
* Moved the layers included in the widget into separate files that implement a `useBoundariesToggleLayer` hook. This was done in an attempt to centralize per-layer logic. More logic could probably be abstracted into the `useBoundariesToggleLayer` hook, but I'd like to add more layers before generalizing too soon.
* Created a _Layers_ context, and restructured the _FetchedData_ context.
    * The _Layers_ context is also an attempt to trim down the _LocationSearch_ context, but since the latter uses classes instead of hooks, it's not easy to integrate yet. The current layers in the _Layers_ context are all `AllFeaturesLayer`s, so they don't need to be reset explicitly, but if we add other layers to that context then we'll need to address resetting them when `resetData` and `setNoDataAvailable` are called for _LocationSearch_.

## Note:
I apologize about the size of this PR, but I'd wanted to wait until all the widget features were added before opening it and also... TypeScript.

## Steps To Test:
1. Currently the widget only has layers that are shown on the **Community** page, so it will be disabled on the **State & Tribal** pages.
2. Navigate to http://localhost:3000/community/dc/overview.
3. Click the "globe" icon widget, and toggle on the **USGS Streamgages**. The visibility of the same layer in the layer list should toggle on as well.
4. Zoom out, and ensure surrounding features are disabled at an appropriate level.
5. Verify the behavior is consistent as tabs are changed. The widget follows the behavior of the `visibleLayers` for each tab.

## TODO:
- [ ] The existing "surrounding" widget layers need to be merged into this one.
- [ ] Per-layer loading bars were discussed. I couldn't find a similar example in the layer list, but remember seeing it in the past.
